setup.py

```diff
@@ -288,6 +315,124 @@ def readme():
 if _has_xenon:
     _ext_mods.append(ext_xenon_wrapper)
 
+# Custom build commands
+#
+# Command to check and adjust code style
+# Usage:
+#    To check style: python ./setup.py style
+#
+#      To fix style: python ./setup.py style -a
+#
+hpat_build_commands = versioneer.get_cmdclass()
+
+
+class style(Command):
+    user_options = [
+        ('apply', 'a', 'Apply codestyle changes to sources.')
+    ]
+    description = "Code style check and apply (with -a)"
+    boolean_options = []
+
+    _result_marker = "Result:"
+    _project_directory_excluded = ['build', '.git']
+
+    _c_formatter = 'clang-format-6.0'
+    _c_formatter_install_msg = 'pip install clang'
```

Comment:
As I looked [here,](https://pypi.org/project/clang/) this command does not install `clang` itself. Where is the code that installs `clang`?

Reply:
You should not pip install anything in the first place. Just import what you need. setup.py should not modify the user's python env. It is possible to import only when really running 'style' so you're not dependent on it if you don't. But I don't see where this is using the clang package anyway.

Reply:
> As I looked [here,](https://pypi.org/project/clang/) this command does not install `clang` itself. Where is the code that installs `clang`?

I had no task to install `clang`. I just looked for clang-format tool installation.

Reply:
> setup.py should not modify the user's python env

It is not implemented in this PR. This line is just help message.

Reply:
Oops...